app/models/course.rb

```diff
@@ -35,7 +35,8 @@ class Course < ActiveRecord::Base
                                 source: :actable, source_type: Course::LessonPlan::Event.name
   has_many :material_folders, class_name: Course::Material::Folder.name, inverse_of: :course,
                               dependent: :destroy
-
+  has_one :root_folder, -> { where(parent_id: nil) },
```

Comment:
@lowjoel there is an alternative solution to resolve the issue above:

```
def root_folder
   material_folders.first || create_root_folder
end
```

It's more reliable, it ensures that every course will return a root folder. But the implementation which I push is more common, what do you think ?